rust/otap-dataflow/crates/otap/src/batch_processor.rs

```diff
@@ -176,29 +310,33 @@ impl Config {
         }
 
         // Presently we have only the OTAP mode of batching, which supports only Items.
-        if self.sizer != Sizer::Items {
+        let (expect_sizer, with_msg) = match format {
+            SignalFormat::OtapRecords => (Sizer::Items, "OTAP batch sizer: must be items"),
+            SignalFormat::OtlpBytes => (Sizer::Bytes, "OTLP batch sizer: must be bytes"),
+        };
+        if self.sizer != expect_sizer {
             return Err(ConfigError::InvalidUserConfig {
-                error: "sizer: must be items".into(),
+                error: with_msg.into(),
             });
         }
 
-        // If both sizes are set, check max_size is >= the batch_size.
-        if let (Some(max_size), Some(batch_size)) = (self.max_size, self.min_size) {
-            if max_size < batch_size {
+        // If both sizes are set, check max_size is >= the min_size.
+        if let (Some(max_size), Some(min_size)) = (self.max_size, self.min_size) {
+            if max_size < min_size {
                 return Err(ConfigError::InvalidUserConfig {
                     error: format!(
                         "max_size ({}) must be >= min_size ({}) or unset",
-                        max_size, batch_size,
+                        max_size, min_size,
                     ),
                 });
             }
         }
 
-        // Zero-timeout is a valid split-only configuration.
-        if self.timeout == Duration::ZERO {
-            if let Some(batch_size) = self.min_size {
+        // Zero-timeout is a valid split-only configuration, but must have
+        if no_timeout {
```

Comment:
⛏️ Did part of this comment get lost? It is unclear to me what the remainder may be.